Template Engine with Subdirectory

nupplaphil opened the pull request:
- Use explicit root path for template engine
- Replace `__DIR__` paramters in addons with subdirectories
- Fixes local tests, where call is made out of '/' and not '/vagrant/'

Needs https://github.com/friendica/friendica-addons/pull/977 as well

MrPetovan commented:
What was wrong with `__DIR__`?

nupplaphil commented:
In some situations (e.g. for tests), a PHP function isn't called out of the friendica base-directory. This happened to me very often in the past when using PHPUnit tests (it led to  template exceptions `didn't find template file XYZ`).

I therefor now use the determined basepath of the application and every template call is now based on the this basepath. So I had to replace `__DIR__` with the relative subpath to make it fully work.

PS: As far as I can see, we only use the 2nd parameter for addons, so we could even pass the addon name instead of the complete subdir :-)

MrPetovan commented:
But `__DIR__` wasn't used in `Renderer::getMarkupTemplate` itself, just as a parameter, so I'm struggling to understand how the template could be missing?

nupplaphil commented:
This is the main change:
https://github.com/friendica/friendica/pull/8545/files#diff-7cc141259626e60dd65e5761cac4cef5R81

And I changed `__DIR__` because it could lead to:
```php
$root = DI::basepath() . '/' . __DIR__;
```
and results in:
`/vagrant/vagrant/view/templates/**`
instead of
`/vagrant/view/templates/**`

Therefor I changed the `$root` parameter to a `$subDir` parameter.

On the other hand, I had the situation where I called PHPUnit at the Vagrant-based virtual image out of `/` instead of the `/vagrant` working directory, which lead to `/view/templates/***` instead of `/vagrant/view/templates/***`. And **this** problem occurred very often in the past (because of the Remote PHP interpreter of PHPStorm and PHPUnit). This PR will solve this issue :-)

MrPetovan commented:
I see, thanks for the elaboration!
